### DISCUSSION TITLE
Disable `tsserver` from initializing entirely

### DISCUSSION THREAD

**Discussion by heysokam** [LazyVim/LazyVim#3308] [Q&A] [answered]
I do not want to use tsserver at all, I use ESLint only.
I uninstall `tsserver` ten times, it reinstalls itself ten other times.
How do I stop this from being automatically added, even though it is not being listed anywhere in my config files?

**Comment by dpetka2001** [comment]
How did you install it? Did you enable it from the Extras? If so you have to disable it from there.

**Reply by heysokam** [reply]
> How did you install it? Did you enable it from the Extras? If so you have to disable it from there.

I don´t remember installing it. So I don´t know the answer to the question, sadly
How can I check?

**Reply by dpetka2001** [reply]
You installed somehow because it's not installed by default. Either through `:LazyExtras` command or in your `/lua/config/lazy.lua` file directly by importing the Extra.

**Reply by heysokam** [reply]
> You installed somehow because it's not installed by default. Either through `:LazyExtras` command or in your `/lua/config/lazy.lua` file directly by importing the Extra.
```lua
    { import = "lazyvim.plugins.extras.lang.typescript" },
```
I found this. But I don't want to disable the rest, only tsserver :thinking:
Is this the option?

**Reply by dpetka2001** [reply]
Will you use a local `tsserver` and don't want the one from Mason? Because there's no reason to keep the rest of the configuration if you uninstall `tsserver`.

**Reply by dpetka2001** [reply, marked as answer]
Just comment out that line and then uninstall `tsserver` from Mason and when you restart Neovim it shouldn't re-install.

**Reply by heysokam** [reply]
> Will you use a local `tsserver` and don't want the one from Mason? Because there's no reason to keep the rest of the configuration if you uninstall `tsserver`.

Idea is to use ESLint, instead of tsserver. I didn't have problems with the rest of the typescript config. Or is tsserver the only thing configured by that import?

**Reply by dpetka2001** [reply]
That import configures everything mentioned [here](https://www.lazyvim.org/extras/lang/typescript).

**Reply by heysokam** [reply]
> Just comment out that line and then uninstall `tsserver` from Mason and when you restart Neovim it shouldn't re-install.

Just tried it, and it did work. tysm :writing_hand:
I had that link you sent on the last message, but for some reason I didn't put two and two together. I though the page was explaining _how_ to configure it, not _what_ is configured. Ty for the clarification :ok_hand:

**Reply by dpetka2001** [reply]
No you don't need to copy anything from the LazyVim website. Those are just the defaults and are shown on the website just for reference. There's also a yellow box saying
> caution
You don't need to copy the default settings to your config. They are only shown here for reference.

**Comment by dustinyoste** [comment]
For anyone coming across this
I found this quick little snippet from this post that worked immediately, no hassle and allows you to just `<leader>lx` to toggle when you don't want the lsps bogging you down
https://samuellawrentz.com/hacks/neovim/disable-annoying-eslint-lsp-server-and-hide-virtual-text/
